### ISSUE TITLE
Array.get() wraps if index out of bounds

### ISSUE THREAD

**Issue by cellularmitosis** [pNre/ExSwift#100]
As a developer becomes familiar with Swift, they will likely find it strange that Apple would provide optional accessors on all of their data structures (which play nicely with the "if let" idiom), with the singular exception of Array.

Instead of being able to do this:

```
if let foo = array[3]
{
    // do something with foo
}
```

Instead you have to do something which feels clumsy in comparison:

```
if array.count > 3
{
    let foo = array[3]!
    // do something with foo
}
```

Writing an optional accessor seems a very obvious thing to do, to make Array fit better with the style of the rest of the language.

So I was quite shocked to discover that your "get()" method automatically wraps an out-of-bounds index, and that this behavior was intentional.  I'm just having trouble wrapping my head around it.  

Who would expect out-of-bounds index wrapping to be the default behavior of a method named "get()"?

If someone needed the (somewhat uncommon) behavior of out-of-bounds index wrapping, why would they want to hide that implementation detail behind a method call, instead of doing something explicit which calls better attention to it, like "array.get(index % array.count)"?

If they insist on hiding the behavior behind a method call, why wouldn't they choose a method name which at least hints at the behavior, like maybe .wrappedGet()?

This behavior and this method name just don't belong together.

**Comment by pNre** [comment]
100% agree, there was probably a reason to have `get` behave this way but I can't figure out what it was.
Since I don't want to add breaking changes to the current version, this is set to be fixed in the next major release.

**Comment by pepaslabs** [comment]
Wonderful!  Thanks.

**Comment by samgreen** [comment]
Let me know if you still think `wrappedGet()` would be a good addition.

**Comment by pNre** [comment]
This issue has been fixed in dd4def6b916bc87e0c3f0861dffd617a6c7733e5.

@samgreen I don't think a `wrappedGet` is needed for now.
